File: ArkhamAsylum.Lib/Models/Building.cs
using Newtonsoft.Json;
using System.Collections.Generic;

namespace ArkhamAsylum.Lib.Models
{
    public class Building : Entity
    {
        public string Code { get; set; }

        public string Address { get; set; }

        [JsonIgnore]
        public ICollection<Floor> Floors { get; set; }
    }
}

File: ArkhamAsylum.Web/wwwroot/Scripts/Models/Building.js
class Building extends Entity
{
    constructor (json)
    {
        super (json)
        if (json)
        {
            this.Code = json.code;
            this.Address = json.address;
        }
        else
        {
            this.Code = "";
            this.Address = "";
        }
    }
}